src/SAML2/XML/saml/Attribute.php

```diff
@@ -129,65 +133,54 @@ public function getFriendlyName(): ?string
      * Set the value of the FriendlyName-property
      *
      * @param string|null $friendlyName
-     * @return void
      */
-    private function setFriendlyName(string $FriendlyName = null): void
+    private function setFriendlyName(?string $friendlyName): void
     {
-        $this->FriendlyName = $FriendlyName;
+        $this->FriendlyName = $friendlyName;
     }
 
 
     /**
      * Collect the value of the attributeValues-property
      *
-     * @return \SAML2\XML\saml\AttributeValue[]|null
+     * @return AttributeValue[]|null
      */
     public function getAttributeValues(): ?array
     {
-        return $this->attributeValues;
+        return $this->AttributeValues;
     }
 
 
     /**
      * Set the value of the AttributeValues-property
      *
      * @param array $attributeValues|null
-     * @return void
      */
-    private function setAttributeValues(?array $attributeValues): void
+    protected function setAttributeValues(?array $attributeValues): void
     {
-        $this->attributeValues = $attributeValues;
-    }
-
-
-    /**
-     * Add the value to the AttributeValues-property
-     *
-     * @param \SAML2\XML\saml\AttributeValue $attributeValue
-     * @return void
-     */
-    public function addAttributeValue(AttributeValue $attributeValue): void
-    {
-        $this->setAttributeValues(
-            empty($this->attributeValues) ? [$attributeValue] : array_merge($this->attributeValues, [$attributeValue])
-        );
+        if ($attributeValues === null) {
+            return;
+        }
+        Assert::allIsInstanceOf($attributeValues, AttributeValue::class, 'Invalid AttributeValue.');
+        $this->AttributeValues = $attributeValues;
     }
 
 
-
     /**
      * Convert XML into a Attribute
      *
-     * @param \DOMElement $xml The XML element we should load
-     * @return \SAML2\XML\saml\Attribute
+     * @param DOMElement $xml The XML element we should load
+     *
+     * @return Attribute
+     * @throws Exception
```

Comment:
Is this your IDE interfering? We use full namespaces in phpdoc throughout this project..
It's especially helpful in really large files, so you don't have to scroll all the way to the top to find out in what namespace `Attribute` exists..

Reply:
Yeah, it marks those as "_unnecessarily qualified_". I can revert those changes, having full namespaced names makes sense in the phpdoc 👍

Thanks for the heads up!